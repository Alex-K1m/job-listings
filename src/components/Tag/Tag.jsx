import cn from 'classnames';
import { func, string } from 'prop-types';
import React from 'react';

import styles from './Tag.module.scss';

const Tag = ({ onClick, onDelete, children }) => {
  return (
    <div className={styles.wrapper}>
      {onClick ? (
        <button
          type='button'
          className={cn(styles.tag, styles.btn)}
          onClick={onClick}
        >
          {children}
        </button>
      ) : (
        <div className={styles.tag}>{children}</div>
      )}
      {onDelete && (
        <button
          type='button'
          className={cn(styles.delete, styles.btn)}
          aria-label={`Remove ${children} tag`}
          onClick={onDelete}
        />
      )}
    </div>
  );
};

Tag.defaultProps = {
  onClick: null,
  onDelete: null,
};

Tag.propTypes = {
  onClick: func,
  onDelete: func,
  children: string.isRequired,
};

export default Tag;
