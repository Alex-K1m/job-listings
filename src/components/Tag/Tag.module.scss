@use '../colors' as c;

.wrapper {
  border-radius: 0.4rem;
  display: inline-flex;
  height: 3.2rem;
  overflow: hidden;
}

.tag {
  --bg-hover: #{c.$primary};
  --color-hover: #fff;
  align-items: center;
  background-color: c.$secondary;
  color: c.$primary;
  display: flex;
  font: 700 1.3rem/185% Spartan, sans-serif;
  height: 100%;
  letter-spacing: -0.0077em;
  padding: 0 0.8rem;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.delete {
  --bg-hover: #{c.$text};
  background: url('../../images/icon-remove.svg') center no-repeat;
  background-color: c.$primary;
  height: 100%;
  width: 3.2rem;
}

.btn {
  border: 0;
  cursor: pointer;
  transition: 0.2s;

  &:focus {
    filter: brightness(0.9);
    outline: none;
  }

  &:hover {
    background-color: var(--bg-hover);
    color: var(--color-hover);
  }
}
