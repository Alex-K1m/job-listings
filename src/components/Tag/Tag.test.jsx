import React from 'react';
import renderer from 'react-test-renderer';

import Tag from './Tag';

describe('Tag', () => {
  const mock = jest.fn();
  const tagName = 'tagName';

  it('renders correctly', () => {
    const tree = renderer.create(<Tag>{tagName}</Tag>);
    const json = tree.toJSON();
    const { root } = tree;

    expect(json).toMatchSnapshot();
    expect(root.findByProps({ className: 'tag' }).children).toEqual([tagName]);
  });

  it('renders correctly with onDelete', () => {
    const tree = renderer.create(<Tag onDelete={mock}>{tagName}</Tag>);
    const json = tree.toJSON();
    const { root } = tree;

    expect(json).toMatchSnapshot();
    expect(
      root.findByProps({ className: 'delete btn' }).props['aria-label'],
    ).toBe(`Remove ${tagName} tag`);
  });

  it('renders correctly with onClick', () => {
    const tree = renderer.create(<Tag onClick={mock}>{tagName}</Tag>);
    const json = tree.toJSON();
    const { root } = tree;

    expect(json).toMatchSnapshot();
    expect(root.findByProps({ className: 'tag btn' }).children).toEqual([
      tagName,
    ]);
  });
});
