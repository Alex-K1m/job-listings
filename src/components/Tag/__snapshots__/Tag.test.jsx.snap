// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="tag"
  >
    tagName
  </div>
</div>
`;

exports[`Tag renders correctly with onClick 1`] = `
<div
  className="wrapper"
>
  <button
    className="tag btn"
    onClick={[MockFunction]}
    type="button"
  >
    tagName
  </button>
</div>
`;

exports[`Tag renders correctly with onDelete 1`] = `
<div
  className="wrapper"
>
  <div
    className="tag"
  >
    tagName
  </div>
  <button
    aria-label="Remove tagName tag"
    className="delete btn"
    onClick={[MockFunction]}
    type="button"
  />
</div>
`;
